.github/workflows/publish-test-results.yml

```diff
@@ -0,0 +1,73 @@
+name: Publish-Test-Results
+
+on:
+  workflow_run:
+    workflows: ["Cloudbeat-CI"]
+    types: 
+      - completed
+jobs:
+  publish_results:
+    name: Publish Results
+    runs-on: ubuntu-latest
+    if: github.event.workflow_run.conclusion != 'skipped'
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    permissions:
+      checks: write
+      # needed unless run with comment_mode: off
+      pull-requests: write
+      # for aws cred access
+      secrets: read
+      # required by download step to access artifacts API
+      actions: read
+    steps:
+      - name: Download and Extract Artifacts
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          mkdir -p tests/allure/results
+          mkdir -p artifacts && cd artifacts
+
+          artifacts_url=${{ github.event.workflow_run.head_sha }}
+
+          gh api "$artifacts_url" -q '.artifacts[] | [.name, .archive_download_url] | @tsv' | while read artifact
+          do
+            IFS=$'\t' read name url <<< "$artifact"
+            gh api $url > "$name.zip"
+            unzip -d "../tests/allure/results" "$name.zip"
+          done
+
+      - name: Publish allure report
+        if: always()
+        uses: andrcuns/allure-publish-action@v1.0.1
+        env:
+          GITHUB_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          storageType: s3
+          resultsGlob: "tests/allure/results/*"
+          updatePr: actions
+          collapseSummary: false
+          summary: suites
+          summaryTableType: markdown
+          copyLatest: true
+          bucket: csp-allure-reports
+          prefix: allure_reports/cloudbeat/${{ github.event.workflow_run.head_branch }}
+          ignoreMissingResults: true
+
+#Todo comment allure results if pr gh ticket https://github.com/elastic/security-team/issues/3936
```

Comment:
this todo is not very clear. Is it referring to a ticket that summarizes what needs to be done? if so we can remove the comment and link this line from the ticket. If not, let's remove it since it is not clear and will probably stay here for a while.

Reply:
Summary with link back is a good idea since this one is critical & complex.

Since tickets get re-shuffled often, I'd like to have that Todo remain there until it's 100% confirmed that  pr input is handled both secure & successfully.

I'll add a more descriptive one-liner & a link back to a detailed summary on the ticket.